src/backend/cuda/mean.cu

```diff
@@ -53,6 +55,8 @@ Array<T> mean(const Array<T>& in, const Array<Tw>& wts, const int dim) {
 
 INSTANTIATE(double, double, double);
 INSTANTIATE(float, float, float);
+INSTANTIATE(half, half, half);
+INSTANTIATE(half, float, half);
```

Comment:
strange but really needed for some other collateral algos : if not then linking issues: cannot find mean<half, float, half>